src/Compilers/CSharp/Portable/Binder/Binder_Operators.cs

```diff
@@ -627,15 +637,25 @@ private void ReportAssignmentOperatorError(AssignmentExpressionSyntax node, Diag
 
         private static void ReportBinaryOperatorError(ExpressionSyntax node, DiagnosticBag diagnostics, SyntaxToken operatorToken, BoundExpression left, BoundExpression right, LookupResultKind resultKind)
         {
-            if ((operatorToken.Kind() == SyntaxKind.EqualsEqualsToken || operatorToken.Kind() == SyntaxKind.ExclamationEqualsToken) &&
-                (left.IsLiteralDefault() && right.IsLiteralDefault()))
+            bool leftDefault = left.IsLiteralDefault();
+            bool rightDefault = right.IsLiteralDefault();
+            if ((operatorToken.Kind() == SyntaxKind.EqualsEqualsToken || operatorToken.Kind() == SyntaxKind.ExclamationEqualsToken))
             {
-                Error(diagnostics, ErrorCode.ERR_AmbigBinaryOpsOnDefault, node, operatorToken.Text);
-                return;
-            }
+                if (leftDefault && rightDefault)
```

Comment:
>leftDefault && rightDefault [](start = 20, length = 27)

Can we simply rely on ```resultKind``` to decide which error to report as we do at the end of this function?